### DISCUSSION TITLE
serve text/html mime type from public bucket urls?

### DISCUSSION THREAD

**Discussion by asafamr** [supabase/supabase#2557] [Questions] [answered]
I love your product!
While developing it could be nice to be able to serve static html straight from a public bucket url.
If I'm not mistaken, content type is overridden and set to plain text when served. (maybe due to users' api same origin security concerns?)
Is there a way to enable text/html mime type serving for a bucket?

**Comment by inian** [comment, marked as answer]
Hi @asafamr, yup we override the content type from html to plain text right now. We do not have a good experience for hosting websites at the moment (no custom domains and SSL cert for custom domains - unless you are okay with hosting your website at https://ref.supabase.co/storage/v1 which I guess most users do not want to). 

However the main reason is that we want to beef up our defences against deceptive webpages and dealing with abuse reports before offering this. This has caused [a lot of issues](https://github.com/supabase/supabase/issues/2096) in the past month. We are well on the way there and once we are more confident about our defences against abuse, we will re-enable hosting webpages on Supabase again.

**Comment by dx3906rocky** [comment]
I got same issue, and finally find the solution. I is need to set the content-type when uploading files to storage.
examples:
        import mimetypes
        file_options = {
            "content-type": content_type
        }

        res = self.client.storage.from_(bucket).upload(
            cloud_file_path, 
            file_path, 
            file_options)

**Reply by inian** [reply]
hi @dx3906rocky, uploading html files is currently not supported. Are you able to serve html files from storage?
